### ISSUE TITLE
Custom train

### ISSUE THREAD

**Issue by Yusepp** [ultralytics/ultralytics#1080]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Is it possible to add a custom dataset that includes images, bounding boxes and facial landmarks and include a custom loss function for the landmarks using the Model/Trainer engine?

**Comment by glenn-jocher** [comment]
@Yusepp yes, that's the idea for this repo! See https://docs.ultralytics.com/engine/

**Comment by Yusepp** [comment]
> @Yusepp yes, that's the idea for this repo! See https://docs.ultralytics.com/engine/

@glenn-jocher so I observed the engine documentation and they say that basically I need to modify the get_model, get_dataloader and Loss functions however:

- There is no simple way to add another data modality such as landmarks without modifying the core model with pytorch. DetectionTrainer and DetectionModel does not provide a simple interface for that.
- It is not clear the format that the dataloader should follow: I observed your dataloader contains a list of dicts and each dict contains the tensors corresponding to each data modality (images, labels, etc) but using pytorch default's dataset/dataloader won't work right?
